frontend/javascripts/admin/dataset/dataset_components.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+import * as React from "react";
+
+import { Form, Input, Select, Card } from "antd";
+import messages from "messages";
+import { isDatasetNameValid } from "admin/admin_rest_api";
+import type { APIDataStore, APIUser } from "admin/api_flow_types";
+
+const FormItem = Form.Item;
+const { Option } = Select;
+
+export function CardContainer({
+  children,
+  withoutCard,
+  title,
+}: {
+  children: React.Node,
+  withoutCard: ?boolean,
+  title: string,
+}) {
+  if (withoutCard) {
+    return <React.Fragment>{children}</React.Fragment>;
+  } else {
+    return (
+      <Card
+        style={{ width: "85%", marginLeft: "auto", marginRight: "auto" }}
+        bordered={false}
+        title={<h3>{title}</h3>}
+      >
+        {children}
+      </Card>
+    );
+  }
+}
+
+export function DatasetNameFormItem({ form, activeUser }: { form: Object, activeUser: ?APIUser }) {
+  const { getFieldDecorator } = form;
+  return (
+    <FormItem label="Dataset Name" hasFeedback>
+      {getFieldDecorator("name", {
+        rules: [
+          { required: true, message: messages["dataset.import.required.name"] },
+          { min: 3 },
+          { pattern: /[0-9a-zA-Z_-]+$/ },
+          {
+            validator: async (_rule, value, callback) => {
+              if (!activeUser) throw new Error("Can't do operation if no user is logged in.");
+              const reasons = await isDatasetNameValid({
+                name: value,
+                owningOrganization: activeUser.organization,
+              });
+              if (reasons != null) {
+                callback(reasons);
+              } else {
+                callback();
+              }
+            },
+          },
+        ],
+        validateFirst: true,
+      })(<Input />)}
+    </FormItem>
+  );
+}
+
+export function DatastoreFormItem({
+  form,
+  datastores,
+}: {
+  form: Object,
+  datastores: Array<APIDataStore>,
+}) {
+  const { getFieldDecorator } = form;
+  return (
+    <FormItem label="Datastore" hasFeedback>
+      {getFieldDecorator("datastore", {
+        rules: [{ required: true, message: messages["dataset.import.required.datastore"] }],
+        initialValue: datastores.length ? datastores[0].url : null,
+      })(
+        <Select
+          showSearch
+          placeholder="Select a Datastore"
+          optionFilterProp="children"
+          style={{ width: "100%" }}
```

Comment:
Maybe add `defaultActiveFirstOption` as a prop so that the first item is autoselected.

Reply:
Did the auto selection of the first item not work for you during testing?
`defaultActiveFirstOption` does not work if `datastores` is `[]` at first and then updated later, it only works on the initial render.
That's why I instead set the `initialValue` in the field decorator (couple of lines below) which worked on subsequent renders as well for me :)

Reply:
Ah, sorry, you are probably right. I reviewed the code before testing :)